statics/src/main/java/com/codeborne/selenide/junit5/ScreenShooterExtension.java

```diff
@@ -61,19 +64,32 @@ public void beforeAll(final ExtensionContext context) {
   }
 
   @Override
-  public void afterEach(final ExtensionContext context) {
+  public void afterAll(final ExtensionContext context) {
+    Screenshots.finishContext();
+  }
+
+  @Override
+  public void testDisabled(ExtensionContext context, Optional<String> reason) {
+
+  }
+
+  @Override
+  public void testSuccessful(ExtensionContext context) {
     if (captureSuccessfulTests) {
       log.info(screenshot(driver()));
-    } else {
-      final Optional<Throwable> executionException = context.getExecutionException();
-      if (executionException.isPresent() && executionException.get() instanceof UIAssertionError) {
-        log.info(screenshot(driver()));
-      }
     }
   }
 
   @Override
-  public void afterAll(final ExtensionContext context) {
-    Screenshots.finishContext();
+  public void testAborted(ExtensionContext context, Throwable cause) {
+
+  }
+
+  @Override
+  public void testFailed(ExtensionContext context, Throwable cause) {
+    if (!(cause instanceof UIAssertionError)) {
```

Comment:
Look like generally you change a logic. In previous implementation we collect expection only if it's instance of `UIAssertionError` but you change this logic. Why?

Reply:
For taking screens for junit assertions. All the details you can see here: #945, and proposed logic will be consistent with other test runners(Junit4, testNG)